Sources/Algorithms/First.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// firstNonNil(of:)
+//===----------------------------------------------------------------------===//
+public extension Sequence {
+    /// Returns the first element in `self` that `transform` maps to a `.some`.
+    func firstNonNil<Result>(of transform: (Element) throws -> Result?) rethrows -> Result? {
```

Comment:
The function name sounds good (as bikeshedded in the ticket), but are we sure about the `of`? Reads a bit weird.
It's more like a `where` that happens to be implemented with `some/none` rather than `true/false` but it's still a form of predicate closer to where rather than "of". 

wdyt?

Reply:
Agreed. I think we can drop the label — to me it reads better when it's more similar to `map`, e.g.: `guard let number = inputs.firstNonNil(Int.init) else { ... }`

Reply:
This feature felt minimal and self-explanatory enough to not be worth a guides doc, but I can add one.